ChiModules/LBSCurvilinear/lbs_curvilinear_sweepchunk_pwl.cc

```diff
@@ -0,0 +1,406 @@
+  const auto fluds = angle_set->fluds;
+
+  const GsSubSet& subset = groupset.grp_subsets[angle_set->ref_subset];
+  const int gs_ss_size  = groupset.grp_subset_sizes[angle_set->ref_subset];
+  const int gs_ss_begin = subset.first;
+  const int gs_gi = groupset.groups[gs_ss_begin].id; // Groupset subset first group number
+
+  int deploc_face_counter = -1;
+  int preloc_face_counter = -1;
+
+  const auto& d2m_op = groupset.quadrature->GetDiscreteToMomentOperator();
+  const auto& m2d_op = groupset.quadrature->GetMomentToDiscreteOperator();
+
+  const auto curvilinear_product_quadrature =
+    std::dynamic_pointer_cast<chi_math::CurvilinearAngularQuadrature>(groupset.quadrature);
+
+  //========================================================== Loop over each cell
+  size_t num_loc_cells = spds->spls.item_id.size();
+  for (spls_index = 0; spls_index < num_loc_cells; ++spls_index)
+  {
+    const int cell_local_id = spds->spls.item_id[spls_index];
+    const auto& cell = grid_view->local_cells[cell_local_id];
+    const auto& fe_intgrl_values = grid_fe_view.GetUnitIntegrals(cell);
+    const auto& fe_intgrl_values_secondary = grid_fe_view_secondary.GetUnitIntegrals(cell);
+    const int num_faces = cell.faces.size();
+    const int num_dofs = fe_intgrl_values.NumNodes();
+    const auto& transport_view = grid_transport_view[cell.local_id];
+    const auto& sigma_tg = xsections[transport_view.xs_id]->sigma_tg;
+    std::vector<bool> face_incident_flags(num_faces, false);
+
+
+    //=================================================== Get Cell matrices
+    const auto& G = fe_intgrl_values.GetIntV_shapeI_gradshapeJ();
+    const auto& M = fe_intgrl_values.GetIntV_shapeI_shapeJ();
+    const auto& N = fe_intgrl_values.GetIntS_shapeI_shapeJ();
+
+    const auto& Maux = fe_intgrl_values_secondary.GetIntV_shapeI_shapeJ();
+
+
+    //=================================================== Loop over angles in set
+    const int ni_deploc_face_counter = deploc_face_counter;
+    const int ni_preloc_face_counter = preloc_face_counter;
+    for (angle_set_index = 0; angle_set_index < angle_set->angles.size(); ++angle_set_index)
+    {
+      deploc_face_counter = ni_deploc_face_counter;
+      preloc_face_counter = ni_preloc_face_counter;
+      const auto& angle_num = angle_set->angles[angle_set_index];
+      const auto& omega = groupset.quadrature->omegas[angle_num];
+
+      const auto polar_level = map_polar_level[angle_num];
+      const auto start_direction = map_start_final[angle_num].first;
+    //const auto final_direction = map_start_final[angle_num].second;
+
+      const auto& fac_diamond_difference =
+        curvilinear_product_quadrature->GetDiamondDifferenceFactor()[angle_num];
+      const auto& fac_streaming_operator =
+        curvilinear_product_quadrature->GetStreamingOperatorFactor()[angle_num];
+
+
+      // ============================================ Gradient matrix
+      for (unsigned int i = 0; i < num_dofs; ++i)
```

Comment:
Since num_dofs is an int, the iteration variable should also be an int.  This applies in multiple places.

Reply:
Rendered consistent in multiple locations.